src/invidious/videos/parser.cr

```diff
@@ -247,8 +268,24 @@ def parse_video_info(video_id : String, player_response : Hash(String, JSON::Any
     .try &.as_a.map &.as_s || [] of String
 
   allow_ratings = video_details["allowRatings"]?.try &.as_bool
+
   family_friendly = microformat["isFamilySafe"]?.try &.as_bool
+  if family_friendly.nil?
+    family_friendly = true # if isFamilySafe not found then assume is safe
+  end
+
   is_listed = video_details["isCrawlable"]?.try &.as_bool
+  if video_badges = video_primary_renderer.try &.dig?("badges")
+    if has_unlisted_badge?(video_badges)
+      is_listed ||= false
+    else
+      is_listed ||= true
+    end
+    # if no badges but videoDetails not available then assume isListed
+  else
+    is_listed ||= true
+  end
```

Comment:
Here you need to check for nil-ness, otherwise these two `is_listed ||= true` will almost always override a previous `false` value.

```suggestion
  if is_listed.nil?
    if video_badges = video_primary_renderer.try &.dig?("badges")
      is_listed = !has_unlisted_badge?(video_badges)
    else
      # If video has no badges and videoDetails is not
      # available, then assume isListed
      is_listed = true
    end
  end
```